doc/pages/expansions.asciidoc

```diff
@@ -218,6 +218,23 @@ The following expansions are supported (with required context _in italics_):
     _in window scope_ +
     line of the main cursor, the third component of `%val{selection_desc}`
 
+*%val{history}*::
+    _in buffer, window scope_ +
+    the full change history of the buffer, including undo forks, as a
+    _history_id_ followed by one or more history entries. _history_id_ may
+    be different from `%val{history_id}` to capture uncommitted changes, for
```

Comment:
I want to know more about these uncommitted changes, and how they get committed. Are they related to Git commits? Is there any point in the program's lifecycle where there are guaranteed to be no uncommitted changes?

I think this means something like "if the user has entered insert mode and begun typing, but not yet hit Escape, _history_id_ may be different from `%val{history_id}` to represent those uncommitted changes."

Reply:
Your last paragraph is correct, it is not related to git.  Maybe the text "changes not yet saved to the history, such typing in insert mode before pressing `<esc>`" would be clearer?

Reply:
I like that, yes!